### PR TITLE
Security fixes and servicemonitor addition

### PR DESCRIPTION
**Pull request by rdavyd**

### CODE REVIEW

**Comment by jcmoraisjr** [comment]
Hi, thanks for the contribution! All the changes are very much appreciated. Following a few suggestions:

* Try to rebase the whole PR to one commit per functionality, so the commit history can tell the story behind the changes and we can merge some of them to older branches. Here a suggested list in no special order:
  * service monitor addition
  * default backend security
  * update default backend version
  * fix label of controller metrics service
  * allow all host paths
* Create new `default-backend-<type>.yaml` files for psp, role, rolebinding and serviceaccount
* HAProxy Ingress doesn't use IngressClass resources (yet, adding in v0.12), is it needed here?
* You can add any number of arguments to the `and` function (`servicemonitor.yaml`, first line)

**Comment by rdavyd** [comment]
Hi,

Grouped some commits together:
- Default backend version bump goes packaged along with security, because 1.0 version does not work w/o root permissions
- Controller PSP fixes/changes packaged as well (apigroupname change, volumes etc.), no reason to split them. PSP didn't work anyway with wrong apigroup naming.
- label of controller metrics service packaged along with servicemonitor, because latter depends on it.

**Comment by rdavyd** [comment]
Updated servicemonitor commit with custom annotations variable.

File: haproxy-ingress/templates/controller-servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+kind: ServiceMonitor
+metadata:
+{{- if .Values.controller.serviceMonitor.annotations }}
+  annotations:
+    {{- toYaml .Values.controller.serviceMonitor.annotations | nindent 4 }}
+{{- end }}
+  labels:
+    {{- include "haproxy-ingress.labels" . | nindent 4 }}
+{{- if .Values.controller.serviceMonitor.labels }}
+    {{ toYaml .Values.controller.serviceMonitor.labels | nindent 4 }}
```
Comment (jcmoraisjr) [inline haproxy-ingress/templates/controller-servicemonitor.yaml:12]: add a trailing `{{-` here to avoid a blank line in the final yaml

**Review by jcmoraisjr — COMMENTED** [review]
One last formatting issue that can be amended in the last commit.

File: haproxy-ingress/templates/controller-servicemonitor.yaml (hunk 1 above)
Comment (rdavyd) [reply haproxy-ingress/templates/controller-servicemonitor.yaml]: Fixed blank line

**Comment by jcmoraisjr** [comment]
Thanks! Merging.

**Pull request merged**